feat(create): auto generate readme on build (#3760)
SAP/open-ux-tools: packages/create/README.md

# `@sap-ux/create` CLI Reference

> Configure features for Fiori applications and projects (0.13.142).

Reviewer: ```suggestion
Configure features for SAP Fiori applications and projects (0.13.142).
```

# [Usage](#usage)

It is recommended to use the cli via `npx` to always get the latest version without the need to install or update it manually. You can run it using:

```sh
npx -y @sap-ux/create@latest [command] [sub-command] /path/to/project
```

If you prefer to download the module, you can install it globally or add it as `devDependency` to a project. Once installed, you can run it using

```sh
# install globally
npm i -g @sap-ux@latest
# or install as devDependency
npm i -D @sap-ux/create@latest
# then run
sap-ux [command] [sub-command] /path/to/project
```

> `Note:` If the project path is not provided, the current working directory will be used.

Author: fixed in d2d1b02eacb7d2819b279c6e0f6328dd726a4e58

---

# [Commands](#commands)

## [` generate`](#-generate)

Generate adaptation projects.

>                     Available Subcommands: adaptation-project

Author: The list of subcommands is dynamic and during generation of the readme the same phrase ("_Available Subcommands: _") is being used for all subcommand lists. Should we go for "Available Subcommand(s)" or something similar? I'm afraid if we use singular here this will not be adjusted once the second entry will be added.

---

> ## [` generate adaptation-project`](#-generate-adaptation-project)

Reviewer: ```suggestion
## [`generate adaptation-project`](#-generate-adaptation-project)
```

> Generate a new UI5 adaptation project with optional prompts and configuration.

Reviewer: Is UI5 correct in this context or should it be SAPUI5?

**Options:**
- `-n, --skip-install` - skip npm install step

> - `-s, --simulate` - simulate only do not write or install

Reviewer: ```suggestion
- `-s, --simulate` - Simulate only. Do not write or install.
```

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

> - `-y, --yes` - use default values for all prompts

Author: fixed

> - `--id [id]` - id of the adaptation project

Reviewer: ```suggestion
- `--id [id]` - The ID of the adaptation project.
```

- `--reference [reference]` - id of the original application

> - `--url [url]` - url pointing to the target system containing the original app

Reviewer: ```suggestion
- `--url [url]` - The URL that points to the target system which contains the original application.
```

Author: fixed

> - `--ignoreCertErrors` - ignore certificate errors when connecting to the target system

Reviewer: ```suggestion
- `--ignoreCertErrors` - Ignore certificate errors when connecting to the target system.
```

Author: fixed

> - `--ft` - enable the Fiori tools for the generated project

Author: fixed

- `--ts` - enable the TypeScript support for the generated project
- `--package [package]` - ABAP package to be used for deployments
- `--transport [transport]` - ABAP transport to be used for deployments

---

## [` add`](#-add)

Add features to an SAP Fiori app.

>                     Available Subcommands: mockserver-config, smartlinks-config, cds-plugin-ui5, inbound-navigation, cards-editor, model, annotations, html, component-usages, deploy-config, variants-config

Author: fixed

---

## [` add mockserver-config`](#-add-mockserver-config)

> Add the necessary configuration for mockserver module @sap-ux/ui5-middleware-fe-mockserver to enable local OData mocking.

Author: fixed

**Options:**
- `-i, --interactive` - ask for config options, otherwise use defaults
- `-n, --skip-install` - skip npm install step
- `-s, --simulate` - simulate only do not write or install; sets also --verbose

> - `-v, --verbose` - show verbose information

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

---

> ## [` add smartlinks-config`](#-add-smartlinks-config)

Author: fixed in d2d1b02eacb7d2819b279c6e0f6328dd726a4e58

> Add a smartLinks configuration to a project for cross-app navigation.

Reviewer: ```suggestion
Add a `smartLinks` configuration to a project for cross-app navigation.
```

Author: fixed

**Options:**

> - `-s, --simulate` - simulate only do not write config; sets also --verbose

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

- `-v, --verbose` - show verbose information

---

## [` add cds-plugin-ui5`](#-add-cds-plugin-ui5)

Add the cds-plugin-ui5 and all prerequisites to a CAP project for UI5 Server integration.

**Options:**
- `-n, --skip-install` - skip npm install step

> - `-s, --simulate` - simulate only, do not write or install; sets also --verbose

Reviewer: ```suggestion
- `-s, --simulate` - Simulate only. Do not write or install. Also sets `--verbose`.
```

> - `-v, --verbose` - show verbose information

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

---

> ## [` add inbound-navigation`](#-add-inbound-navigation)

Author: fixed in d2d1b02eacb7d2819b279c6e0f6328dd726a4e58

> Add Fiori Launchpad inbound navigation configuration to a project.

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

**Options:**

> - `-s, --simulate` - simulate only do not write config; sets also --verbose

Reviewer: ```suggestion
- `-s, --simulate` - Simulate only. Do not write to the config file. Also, sets `--verbose`.
```

- `-v, --verbose` - show verbose information

> - `-c, --config <string>` _(required)_ - Path to project configuration file in YAML format _(default: "ui5.yaml")_

Reviewer: ```suggestion
- `-c, --config <string>` _(required)_ - Path to the project configuration file in YAML format. _(default: `ui5.yaml`)_
```

---

## [` add cards-editor`](#-add-cards-editor)

Add the necessary configuration to an existing yaml file and the script to package.json for cards generation. It will use the configuration from the yaml file passed by cli or default to ui5.yaml, as provided by the fiori-tools-preview or preview-middleware.

**Options:**

> - `-c, --config <string>` _(required)_ - Path to project configuration file in YAML format _(default: "ui5.yaml")_

Author: fixed

- `-s, --simulate` - simulate only do not write config; sets also --verbose
- `-v, --verbose` - show verbose information

---

## [` add model`](#-add-model)

> Add a new OData service and UI5 model to an existing adaptation project.

Reviewer: Is UI5 correct? Should it be SAPUI5 or does this apply to both OpenUI5 and SAPUI5?

Author: Good question. actually it's an odata model. Let me check

**Options:**
- `-s, --simulate` - simulate only do not write or install

---

## [` add annotations`](#-add-annotations)

Adds an annotation to the OData Source of the base application in an adaptation project.

**Options:**

> - `-s, --simulate` - simulate only do not write or install

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

- `-c, --config <string>` _(required)_ - Path to project configuration file in YAML format _(default: "ui5.yaml")_

---

## [` add html`](#-add-html)

> Add html files for local preview and testing to the project. It will use the configuration from the `ui5.yaml` as default, as provided by the `fiori-tools-preview` or `preview-middleware`.

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

**Options:**
- `-c, --config <string>` _(required)_ - Path to project configuration file in YAML format _(default: "ui5.yaml")_
- `-s, --simulate` - simulate only do not write config; sets also --verbose

> - `-v, --verbose` - show verbose information

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

---

> ## [` add component-usages`](#-add-component-usages)

Reviewer: ```suggestion
## [`add component-usages`](#-add-component-usages)
```

Author: fixed in d2d1b02eacb7d2819b279c6e0f6328dd726a4e58

Add the component usages to an adaptation project.

**Options:**
- `-s, --simulate` - simulate only do not write or install

---

> ## [` add deploy-config`](#-add-deploy-config)

Reviewer: ```suggestion
## [`add deploy-config`](#-add-deploy-config)
```

Prompt for ABAP deployment configuration details and add/update the project files accordingly.

**Options:**
- `-t, --target <string>` _(required)_ - target for deployment; ABAP or Cloud Foundry (not yet implemented)
- `-s, --simulate` - simulate only do not write; sets also --verbose
- `-v, --verbose` - show verbose information
- `-b, --base-file <string>` _(required)_ - the base file config file of the project; default : ui5.yaml

> - `-d, --deploy-file <string>` _(required)_ - the name of the deploy config file to be written; default : ui5-deploy.yaml

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

---

> ## [` add variants-config`](#-add-variants-config)

Reviewer: ```suggestion
## [`add variants-config`](#-add-variants-config)
```

Add the necessary configuration to an existing yaml file and the script to package.json for variants creation. It will use the configuration from the yaml file passed by cli or default to `ui5.yaml`, as provided by the `fiori-tools-preview` or `preview-middleware`.

**Options:**

> - `-c, --config <string>` _(required)_ - Path to project configuration file in YAML format _(default: "ui5.yaml")_

Author: fixed

- `-s, --simulate` - simulate only do not write config; sets also --verbose
- `-v, --verbose` - show verbose information

---

## [` convert`](#-convert)

Convert existing SAP Fiori applications.
                    Available Subcommands: preview-config

---

> ## [` convert preview-config`](#-convert-preview-config)

Reviewer: ```suggestion
## [`convert preview-config`](#-convert-preview-config)
```

> Executed in the root folder of an app, it will convert the respective app to the preview with virtual endpoints. It will use the configuration from the scripts in the `package.json` file to adjust the UI5 configuration YAML files accordingly. The obsolete JS and TS sources will be deleted and the HTML files previously used for the preview will be renamed to `*_old.html`.

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

**Options:**

> - `-s, --simulate <boolean>` _(required)_ - simulate only do not write

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

- `-v, --verbose` - show verbose information

> - `-t, --tests <boolean>` _(required)_ - also convert test suite and test runners

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

---

> ## [` remove`](#-remove)

Reviewer: ```suggestion
## [`remove`](#-remove)
```

Remove features from existing SAP Fiori applications.
                    Available Subcommands: mockserver-config

---

> ## [` remove mockserver-config`](#-remove-mockserver-config)

Reviewer: ```suggestion
## [`remove mockserver-config`](#-remove-mockserver-config)
```

Author: fixed in d2d1b02eacb7d2819b279c6e0f6328dd726a4e58

> Removes the configuration for mockserver module @sap-ux/ui5-middleware-fe-mockserver.

Reviewer: ```suggestion
Removes the configuration for the `@sap-ux/ui5-middleware-fe-mockserver` mockserver module.
```

**Options:**

> - `-v, --verbose` - show verbose information

Reviewer: ```suggestion
- `-v, --verbose` - Show verbose information.
```

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

> - `-f, --force` - do not ask for confirmation when deleting files

Reviewer: ```suggestion
- `-f, --force` - Do not ask for confirmation when deleting files.
```

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

---

## [` change`](#-change)

Change existing adaptation projects.

>                     Available Subcommands: data-source, inbound

Author: fixed

---

## [` change data-source`](#-change-data-source)

Replace the OData Source of the base application in an adaptation project.

**Options:**

> - `-s, --simulate` - simulate only do not write or install

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8

> - `-c, --config <string>` _(required)_ - Path to project configuration file in YAML format _(default: "ui5.yaml")_

Author: fixed

---

> ## [` change inbound`](#-change-inbound)

Author: fixed in d2d1b02eacb7d2819b279c6e0f6328dd726a4e58

> Replace the Inbound FLP configurations of the base application in an adaptation project.

Reviewer: ```suggestion
Replace the inbound FLP configurations of the base application in an adaptation project.
```

**Options:**

> - `-s, --simulate` - simulate only do not write or install

Author: fixed in 57c5e7dc6c96ce92113f2f9bc7a945d8feea29d8
